rust_src/src/fonts.rs

```diff
@@ -265,4 +310,51 @@ pub fn font_at_lisp(
     unsafe { font_at(-1, pos, ptr::null_mut(), w.as_mut(), string) }
 }
 
+/// List available fonts matching FONT-SPEC on the current frame.
+/// Optional 2nd argument FRAME specifies the target frame.
+/// Optional 3rd argument NUM, if non-nil, limits the number of returned fonts.
+/// Optional 4th argument PREFER, if non-nil, is a font-spec to
+/// control the order of the returned list.  Fonts are sorted by
+/// how close they are to PREFER.
+#[lisp_fn(min = "1")]
+pub fn list_fonts(
+    font_spec: LispFontSpecRef,
+    frame: LispFrameLiveOrSelected,
+    num: Option<LispNumber>,
+    prefer: Option<LispFontSpecRef>,
+) -> LispObject {
+    let mut frame: LispFrameRef = frame.into();
+
+    let n = match num.map(|n| n.to_fixnum()) {
```

Comment:
Is there a `to_maybe_fixnum` conversion or something like that?

Reply:
Assuming brotzeit is right and `num` is supposed to be an EmacsInt, the map won't be necessary

Reply:
in lisp.h `#define lisp_h_CHECK_NUMBER(x) CHECK_TYPE (INTEGERP (x), Qintegerp, x)` yep, that is a EmacsInt.